autoPyTorch/datasets/base_dataset.py

```diff
@@ -146,7 +146,7 @@ def __init__(
 
         # TODO: Look for a criteria to define small enough to preprocess
         # False for the regularization cocktails initially
-        self.is_small_preprocess = False
+        # self.is_small_preprocess = False
```

Comment:
What is `small_preprocess`?

Reply:
So during last week's meeting we discussed about how to handle preprocessing in regularisation cocktails. If you remember, we had moved preprocessing to the tabular feature validator and disabled their hyperparameters. But we decided that we need to search the hyperparameters for preprocessing as well. So, in this PR I am making preprocessing back to how it is in development. Moreover, during our experiments for regularisation cocktails, we realised that a lot of time is taken due to transforming the dataset per epoch as it is currently implemented in the development branch. It's better to preprocess the whole dataset all at once per configuration. The idea behind `is_small_preprocess` was to use it as a flag that can be used to determine if the dataset can be preprocessed all at once or not. Based on this we either preprocessed the dataset in the early preprocessing node, or we move these to the transforms in the data loader. However, we never actually used this flag and always added it to the data loader transforms. With this PR, in conjunction with what we discussed in the meeting, I have removed this flag and now we preprocess the whole dataset in the early preprocessing node.

Reply:
Thanks, I will copy your comment to google doc.